vllm/v1/attention/backends/mla/common.py

```diff
@@ -411,26 +434,38 @@ def __init__(self,
                 # which would result in up-projected context being
                 #   2*(192*128)*(64*1024) = 3gb
                 # (assuming 192 QK head dim, 128 heads, and fp16)
-                128 * 1024)
-            assert self.chunked_prefill_workspace_size >= \
-                scheduler_config.max_num_seqs * cache_config.block_size
+                128 * 1024,
+            )
+            assert (self.chunked_prefill_workspace_size
+                    >= scheduler_config.max_num_seqs * cache_config.block_size)
             self.chunked_prefill_workspace = torch.empty(
-                (self.chunked_prefill_workspace_size,
-                 model_config.get_head_size()),
+                (
+                    self.chunked_prefill_workspace_size,
+                    model_config.get_head_size(),
+                ),
                 dtype=model_config.dtype,
                 device=runner.device,
             )
+
+            self.cudnn_workspace = torch.empty(
+                CUDNN_WORKSPACE_SIZE * scheduler_config.max_num_seqs,
+                dtype=torch.int8,
+                device=runner.device,
+            ) if use_cudnn_prefill() else None
+
         self.block_table = block_table
 
         self._use_fi_prefill = use_flashinfer_prefill()
-        self.prefill_metadata_cls = FlashInferPrefillMetadata \
-            if self._use_fi_prefill else MLACommonPrefillMetadata
+        self.prefill_metadata_cls = (FlashInferPrefillMetadata
+                                     if self._use_fi_prefill else
+                                     MLACommonPrefillMetadata)
```

Comment:
Do we need FlashInferPrefillMetadata for both `use_flashinfer_prefill() or use_cudnn_prefill()`?